Helpers/get_results_from_criteria_and_data.py

```diff
@@ -0,0 +1,78 @@
+from datetime import date, timedelta
+from typing import List
+
+from graphql_objects import DatesCheck
+
+
+class GetResultsFromSearchCriteriaAndData():
+
+    @staticmethod
+    def get_data_check_results(validated_search_criteria, id_houses_dict: dict) -> List[DatesCheck]:
+        results_list: List[DatesCheck] = []
+        checkin_desired = validated_search_criteria.desired_checkin
+        checkout_desired = validated_search_criteria.desired_checkout
+        for house in id_houses_dict.values():
+            data_check = DatesCheck(listing_id=house.listing_id,
+                                    is_available=True,
+                                    checkin_date=checkin_desired,
+                                    checkout_date=checkout_desired)
+            if checkin_desired is None and checkout_desired is None:
+                data_check = GetResultsFromSearchCriteriaAndData.update_data_check_when_null_checkin_checkout(
+                    data_check= data_check,
+                    house= house)
+            elif checkout_desired is None:
+                data_check = GetResultsFromSearchCriteriaAndData.update_data_check_when_only_non_null_checkin(
+                    checkin_desired=checkin_desired,
+                    data_check=data_check,
+                    house=house,
+                )
+            else:
+                data_check = GetResultsFromSearchCriteriaAndData.update_data_check_when_non_null_checkin_checkout(
+                    checkin_desired=checkin_desired,
+                    checkout_desired=checkout_desired,
+                    data_check=data_check,
+                    house=house,
+                    )
+            results_list.append(data_check)
+
+            if len(results_list) == 0:
+                data_check.is_available = False
```

Comment:
Very hacky to initialise a `DatesCheck` object with `True` first, and then set it to `False` in this and the other helper functions. I would only initialize the `DatesCheck` when I have the correct values for it.